zebrad/src/commands/start/sync.rs

```diff
@@ -227,31 +230,37 @@ where
                 .map_err(|e| eyre!(e))?
                 .call(zn::Request::BlocksByHash(set));
 
-            let mut verifier = self.verifier.clone();
+            let verifier = self.verifier.clone();
 
             let _ = tokio::spawn(async move {
-                match async move {
+                let result_fut = async move {
+                    let mut handles = FuturesUnordered::new();
```

Comment:
I'm unsure about the error and result propagation in this logic, but I would prefer to try to fix it up in a different issue/PR than this one.

Reply:
Okay, fwiw this is intentionally half complete, we have to store the handle from the outer spawn to ever propagate the errors back to the `run` method, but I was going to deal with that in a later PR.